backend/booking/views.py

```diff
@@ -572,3 +572,113 @@ def getHighRatedClubs(request):
+
+
+from django.utils.dateparse import parse_datetime
+from django.db.models import Q
+
+@api_view(['GET'])
+def getNearestSlot(request):
+    court_id = request.query_params.get('courtId')
+    date_str = request.query_params.get('date')
+    
+    date_obj = parse_datetime(date_str)
+    current_datetime = datetime.datetime.now().replace(microsecond=0)
+    current_date = current_datetime.date()
+    selected_date = date_obj.date() if date_obj else current_datetime.date()
+    if (selected_date == current_date):
+        current_time = (current_datetime + datetime.timedelta(hours=1)).time()
+
+    # Exclude unavailable and booked slots
+    unavailable_slots = UnavailableSlot.objects.filter(
+        court_id=court_id, 
+        date__gte=selected_date, 
+        is_active=True
+    ).values_list('start_time', 'end_time')
+    
+    booked_slots = Booking.objects.filter(
+        court_id=court_id, 
+        booking_date__gte=selected_date
+    ).values_list('slot__start_time', 'slot__end_time')
+    
+    excluded_times = list(unavailable_slots) + list(booked_slots)
+
+    nearest_slots = []
+
+    # Find nearest slot from Slot table
+    for i in range(0, 7):  # Search for up to one week
+        target_date = selected_date + timedelta(days=i)
+        target_weekday = target_date.strftime('%A')
+
+        slots = Slot.objects.filter(
+            court_id=court_id,
+            days=target_weekday,
+            start_time__gte=(current_time if i == 0 and selected_date == current_datetime.date() else datetime.datetime.min.time())
+        ).exclude(
+            Q(start_time__in=[time[0] for time in excluded_times]) |
+            Q(end_time__in=[time[1] for time in excluded_times])
+        )
+
+        for slot in slots:
+            slot_info = {
+                'date': target_date,
+                'start_time': slot.start_time,
+                'end_time': slot.end_time,
+                'source': 'slot'
+            }
+            nearest_slots.append(slot_info)
+        
+    # Find nearest slot from AdditionalSlot table
+    for i in range(0, 7):  # Search for up to one week
+        target_date = selected_date + timedelta(days=i)
+
+        additional_slots = AdditionalSlot.objects.filter(
+            court_id=court_id,
+            date=target_date,
+            is_active=True,
+            start_time__gte=(current_time if i == 0 and selected_date == current_datetime.date() else datetime.datetime.min.time())
+        ).exclude(
+            Q(start_time__in=[time[0] for time in excluded_times]) |
+            Q(end_time__in=[time[1] for time in excluded_times])
+        )
+
+        for slot in additional_slots:
+            slot_info = {
+                'date': slot.date,
+                'start_time': slot.start_time,
+                'end_time': slot.end_time,
+                'source': 'additional_slot'
+            }
+            nearest_slots.append(slot_info)
```

Comment:
Seem this block of code is repeating in multiple places, think of reducing duplicates.